examples/Reranking_search_results_with_cross-encoders.ipynb

```diff
@@ -0,0 +1,805 @@
+    "Relevant to the query?\n",
+    "Assistant: Yes\n",
+    "\n",
+    "Query: Explain sentence embeddings\n",
+    "Retrieved document: Inside the bubble: exploring the environments of reionisation-era Lyman-α emitting galaxies with JADES and FRESCO: We present a study of the environments of 16 Lyman-α emitting galaxies (LAEs) in the reionisation era (5.8<z<8) identified by JWST/NIRSpec as part of the JWST Advanced Deep Extragalactic Survey (JADES). Unless situated in sufficiently (re)ionised regions, Lyman-α emission from these galaxies would be strongly absorbed by neutral gas in the intergalactic medium (IGM). We conservatively estimate sizes of the ionised regions required to reconcile the relatively low Lyman-α velocity offsets (ΔvLyα<300kms−1) with moderately high Lyman-α escape fractions (fesc,Lyα>5%) observed in our sample of LAEs, indicating the presence of ionised ``bubbles'' with physical sizes of the order of 0.1pMpc≲Rion≲1pMpc in a patchy reionisation scenario where the bubbles are embedded in a fully neutral IGM. Around half of the LAEs in our sample are found to coincide with large-scale galaxy overdensities seen in FRESCO at z∼5.8-5.9 and z∼7.3, suggesting Lyman-α transmission is strongly enhanced in such overdense regions, and underlining the importance of LAEs as tracers of the first large-scale ionised bubbles. Considering only spectroscopically confirmed galaxies, we find our sample of UV-faint LAEs (MUV≳−20mag) and their direct neighbours are generally not able to produce the required ionised regions based on the Lyman-α transmission properties, suggesting lower-luminosity sources likely play an important role in carving out these bubbles. These observations demonstrate the combined power of JWST multi-object and slitless spectroscopy in acquiring a unique view of the early stages of Cosmic Reionisation via the most distant LAEs.\n",
+    "Relevant to the query?\n",
+    "Assistant: No\n",
+    "\"\"\"\n",
+    "End of Examples\n",
+    "\n",
+    "Begin!\n",
+    "\n",
+    "Query: {query}\n",
+    "Retrieved document: {document}\n",
+    "Relevant to the query?\n",
+    "Assistant:\n",
+    "'''\n",
+    "\n",
+    "\n",
+    "@retry(wait=wait_random_exponential(min=1, max=40), stop=stop_after_attempt(3))\n",
+    "def rerank_document(query, document, output_list):\n",
```

Comment:
not a big deal, but one way to do this function would be to have it return the output list and then do the appending outside of the function. I tend to prefer functions that return a value rather than mutate the state of an input (feels easier for readers to follow), but either way works.

Reply:
Refactored this one to return a tuple which we can append outside.